File: C#/Classes/ScoreKeeper.cs
using System.Collections.Generic;
using System.Linq;

namespace Singleton.Classes
{
	public class ScoreKeeper
	{
		#region Singleton Setup
		private static ScoreKeeper _scoreKeeper;

		// This is important for this pattern
		// You do not want users to create instances of this class
		// Instead, they should access the class via the GetInstance method
		private ScoreKeeper()
		{
			_scores = new Dictionary<string, int>();
		}

		public static ScoreKeeper GetInstance()
		{
			if (_scoreKeeper == null)
				_scoreKeeper = new ScoreKeeper();

			return _scoreKeeper;
		}
		#endregion

		private IDictionary<string, int> _scores;

		public void IncrementScore(string playerName)
		{
			if (!_scores.Keys.Any(x => x == playerName))
				_scores.Add(playerName, 0);

			_scores[playerName] += 1;
		}

		public IDictionary<string, int> GetScores()
		{
			return _scores;
		}
	}
}

File: Python/ScoreKeeper.py
class ScoreKeeper:

    def __init__(self):
        self.scores = {}

    '''BEGIN Singleton Setup'''
    __instance = None

    @staticmethod
    def get_instance():
        if not ScoreKeeper.__instance:
            ScoreKeeper.__instance = ScoreKeeper()
        return ScoreKeeper.__instance
    '''END Singleton Setup'''

    def increment_score(self, player_name):
        if player_name not in self.scores:
            self.scores[player_name] = 0
        self.scores[player_name] = self.scores[player_name] + 1

    def get_scores(self):
        return self.scores
